lib/compress/zstd_compress.c

```diff
@@ -5186,7 +5378,10 @@ static ZSTD_compressionParameters ZSTD_dedicatedDictSearch_getCParams(int const
 static int ZSTD_dedicatedDictSearch_isSupported(
         ZSTD_compressionParameters const* cParams)
 {
-    return (cParams->strategy >= ZSTD_greedy) && (cParams->strategy <= ZSTD_lazy2);
+    return (cParams->strategy >= ZSTD_greedy)
+        && (cParams->strategy <= ZSTD_lazy2)
+        && (cParams->hashLog >= cParams->chainLog)
+        && (cParams->chainLog <= 24);
```

Comment:
This seems completely unrelated to the row based match finder?

If this change is necessary, can you please put it in a separate PR with its own test case exposing why it is needed?

Reply:
#2540 -> this PR would fail unit tests without the fix back when we added the extra strategies, but it makes sense to separate it out